### ISSUE TITLE
RustDesk Pro - installation automation

### ISSUE THREAD

**Issue by DMSnz** [rustdesk/rustdesk-server-pro#20]
As the title suggests, looking for best ideas to include some form of installation automation for the masses.
NOTE: for the Pro version.

I would propose without having much experience;
Parameter based install options that can be thrown out via PowerShell or GPO etc.
Parameters should/could be stored in a central config file that can be queried at install

Other thoughts / better ideas welcomed please, but a bare bones option to those without any package installers like Puppet, Chocolatey and so on would be beneficial.

**Comment by rustdesk** [comment]
👍 I like your proposal.

**Comment by rustdesk** [comment]
@dinger1986 @it-boerner

**Comment by rustdesk** [comment]
> Parameter based install options that can be thrown out via PowerShell or GPO etc.
> Parameters should/could be stored in a central config file that can be queried at install
> EDIT: To allow a variable based username to show up in the Device List via Web Console would be great (so at install, can specify %username% for a Windows variable at least)

Could you let me know how do you run the PowerShell? A snapshot will help a lot.

**Comment by DMSnz** [comment]
> > Parameter based install options that can be thrown out via PowerShell or GPO etc.
> > Parameters should/could be stored in a central config file that can be queried at install
> > EDIT: To allow a variable based username to show up in the Device List via Web Console would be great (so at install, can specify %username% for a Windows variable at least)
> 
> Could you let me know how do you run the PowerShell? A snapshot will help a lot.

I have maybe jumped the gun on this...
In reality to install exe via a PowerShell script (remotely on various machines) would at the very least need local administrator rights if running said script via GPO, if running directly remotely you need services running to allow the remote connection which are not running out of the box.
I believe GPO based "Software deployment" installs only work with MSI files not EXE files

There are various workarounds for these issues but mostly hack jobs and not therefore a valid option for most.
Back to thinking about this...

**Comment by dinger1986** [comment]
I have added alot of this in the PR https://github.com/rustdesk/doc.rustdesk.com/pull/232

I did a write up for Tactical RMM https://docs.tacticalrmm.com/3rdparty_rustdesk/ and will be including alot of this info in the FAQs

Also we reference it here https://github.com/rustdesk/rustdesk-server-pro/issues/16

**Comment by dinger1986** [comment]
this should help as well https://github.com/rustdesk/rustdesk/commit/9ac3128309fa9812016fd7c3ce43e40d3c24003f#r122060835 if we can import the config via a string which is available on the web console of the api then deployment is very simple going forward.

Currently the full deployment powershell script is here https://rustdesk.com/docs/en/self-host/pro/faq/#can-i-deploy-rustdesk-using-powershell I used it to deploy over 400 agents last night

Powershell can be used effectively with GPO and intune etc

**Comment by DMSnz** [comment]
Dead keen to try this
But would like to wait/see how we can integrate username/device name in to the console before 200 odd machines report back to the console with ID number only.

**Comment by it-boerner** [comment]
Thanks for your great work incorporating the deployment commands for RustDesk via PowerShell

**Comment by rustdesk** [comment]
> Dead keen to try this But would like to wait/see how we can integrate username/device name in to the console before 200 odd machines report back to the console with ID number only.

Yes, working on this.

**Comment by DMSnz** [comment]
> > Dead keen to try this But would like to wait/see how we can integrate username/device name in to the console before 200 odd machines report back to the console with ID number only.
> 
> Yes, working on this.

All good no pressure, appreciating the fast paced dev

**Comment by dinger1986** [comment]
Yes would need to get a flag for that on the exe first but once done the script would work.

Yeah TRMM can run as system or as user works really well with RustDesk Server Pro

**Comment by rustdesk** [comment]
> Only thing I'd add to this request is that the installation scripts automatically adds the endpoint to the address book.

We will think about this, or some other alternative.

**Comment by dinger1986** [comment]
> this should help as well [rustdesk/rustdesk@9ac3128#r122060835](https://github.com/rustdesk/rustdesk/commit/9ac3128309fa9812016fd7c3ce43e40d3c24003f#r122060835) if we can import the config via a string which is available on the web console of the api then deployment is very simple going forward.

This does indeed work
